controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -423,7 +425,18 @@ func (r *KubeadmControlPlaneReconciler) generateKubeadmConfig(ctx context.Contex
 	return bootstrapRef, nil
 }
 
+func (r *KubeadmControlPlaneReconciler) pickFailureDomain(ctx context.Context, kcp *controlplanev1.KubeadmControlPlane, cluster *clusterv1.Cluster) string {
+	machineList, err := r.getMachines(ctx, types.NamespacedName{Namespace: cluster.GetNamespace(), Name: cluster.GetName()})
+	if err != nil {
+		r.Log.Info("Failure domain is not being used due to a kubernetes client error", "error", err)
+		return ""
```

Comment:
I'm wondering if it might be better to not swallow this error and bubble it up, so that we don't make poor scaling decisions just because we could not list machines.

As an aside, do we need to take into account the case of no machines here and swallow a "NotFound" error?

Reply:
hmm, I don't quite follow. I'm approaching this from the perspective that if we can't pick failure domains then we can't pick failure domains and not setting a failure domain is fine. Should this be a strict requirement and we should fail if we can't pick a failure domain? Not sure what you mean by making a poor scaling decision.